src/plugins/controls/public/control_group/external_api/control_group_renderer.tsx

```diff
@@ -6,136 +6,176 @@
 
+import { CONTROL_GROUP_TYPE } from '../../../common';
 import {
-  getDefaultControlGroupInput,
-  getDefaultControlGroupPersistableInput,
-  persistableControlGroupInputKeys,
-} from '../../../common';
-import { ControlGroupContainer } from '../embeddable/control_group_container';
-import { ControlGroupContainerFactory } from '../embeddable/control_group_container_factory';
+  ControlGroupApi,
+  ControlGroupRuntimeState,
+  ControlGroupSerializedState,
+} from '../../react_controls/control_group/types';
 import {
-  ControlGroupCreationOptions,
-  ControlGroupInput,
-  ControlGroupOutput,
-  CONTROL_GROUP_TYPE,
-} from '../types';
+  type ControlGroupStateBuilder,
+  controlGroupStateBuilder,
+} from '../../react_controls/control_group/utils/control_group_state_builder';
 import {
   AwaitingControlGroupAPI,
-  buildApiFromControlGroupContainer,
-  ControlGroupAPI,
-} from './control_group_api';
-import { controlGroupInputBuilder, ControlGroupInputBuilder } from './control_group_input_builder';
+  ControlGroupCreationOptions,
+  ControlGroupRendererApi,
+} from './types';
+import { getDefaultControlGroupRuntimeState } from '../../react_controls/control_group/utils/initialization_utils';
 
 export interface ControlGroupRendererProps {
-  filters?: Filter[];
   getCreationOptions?: (
-    initialInput: Partial<ControlGroupInput>,
-    builder: ControlGroupInputBuilder
-  ) => Promise<ControlGroupCreationOptions>;
+    initialState: Partial<ControlGroupRuntimeState>,
+    builder: ControlGroupStateBuilder
+  ) => Promise<Partial<ControlGroupCreationOptions>>;
+  viewMode?: ViewModeType;
+  filters?: Filter[];
   timeRange?: TimeRange;
   query?: Query;
 }
 
 export const ControlGroupRenderer = forwardRef<AwaitingControlGroupAPI, ControlGroupRendererProps>(
-  ({ getCreationOptions, filters, timeRange, query }, ref) => {
-    const [controlGroup, setControlGroup] = useState<ControlGroupContainer>();
+  ({ getCreationOptions, filters, timeRange, query, viewMode }, ref) => {
+    const id = useMemo(() => uuidv4(), []);
+    const [regenerateId, setRegenerateId] = useState(uuidv4());
+
+    const [apiLoading, setApiLoading] = useState<boolean>(true);
+    const [controlGroup, setControlGroup] = useState<ControlGroupRendererApi | undefined>();
+    useImperativeHandle(ref, () => controlGroup as ControlGroupRendererApi, [controlGroup]);
+
+    const searchApi = useSearchApi({
+      filters,
+      query,
+      timeRange,
+    });
+    const viewMode$ = useMemo(
+      () => new BehaviorSubject<ViewModeType>(viewMode ?? ViewMode.VIEW),
+      // viewMode only used as initial value - changes do not effect memoized value.
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      []
+    );
+    useEffect(() => {
+      if (viewMode) viewMode$.next(viewMode);
+    }, [viewMode, viewMode$]);
+
+    const reload$ = useMemo(() => new BehaviorSubject<void>(undefined), []);
```

Comment:
`BehaviorSubject` expects an initial value - I get the following warning when I tried this:

```
Expected 1 arguments, but got 0.ts(2554)
BehaviorSubject.d.ts(10, 17): An argument for '_value' was not provided.
```

Reply:
reload should be typed as just `Subject`. Then it can be created like 

https://github.com/elastic/kibana/blob/main/examples/controls_example/public/app/react_control_example/react_control_example.tsx#L102

Reply:
Oh!! Of course 🤦 Thanks, hahah - skipped over that it was `Subject` and not `BehaviorSubject`.  Done in 871b82698b96b526ac1d5c59666765f222503163